### PR TITLE
Don't return error in ContextFromRequest function.

### PR DESCRIPTION
**Pull request by pstibrany**
Previously `httputil.ContextFromRequest` function could return error if `http.Request.RemoteAddr` field didn't have correct format, but since this field explicitly has no defined format, that was little too strict.

We have run into this issue when integrating [Prometheus 2.16.0 into Cortex](https://github.com/cortexproject/cortex/pull/2088).

### CODE REVIEW

File: util/httputil/context.go [hunk 1]
```diff
@@ -33,20 +33,24 @@ func ContextWithPath(ctx context.Context, path string) context.Context {
 
 // ContextFromRequest returns a new context from a requests with identifiers of
 // the request to be used later when logging the query.
-func ContextFromRequest(ctx context.Context, r *http.Request) (context.Context, error) {
-	ip, _, err := net.SplitHostPort(r.RemoteAddr)
-	if err != nil {
-		return ctx, err
+func ContextFromRequest(ctx context.Context, r *http.Request) context.Context {
+	m := map[string]string{
```
Comment (bwplotka) [inline util/httputil/context.go]: why `m`? `reqCtxVal` maybe to be more explicit?

Comment (pstibrany) [reply util/httputil/context.go]: Given how small the function is, `m` looked enough to me, but I can change it to `reqCtxVal` as well.

File: util/httputil/context.go [hunk 2]
```diff
@@ -33,20 +33,24 @@ func ContextWithPath(ctx context.Context, path string) context.Context {
 
 // ContextFromRequest returns a new context from a requests with identifiers of
```
Comment (simonpasquier) [inline util/httputil/context.go]: LGTM. Can you just fix this comment?

```suggestion
// ContextFromRequest returns a new context from a request with identifiers of
```

Comment (pstibrany) [reply util/httputil/context.go]: what about?

```suggestion
// ContextFromRequest returns a new context with identifiers of 
// the request to be used later when logging the query.
```

**Comment by roidelapluie** [comment]
Can you please fix DCO?

**Comment by pstibrany** [comment]
> Can you please fix DCO?

Done. Thanks, missed the last commit.

**Comment by roidelapluie** [comment]
Is there a real-life situation where that can error?

**Comment by roidelapluie** [comment]
This looks strange to me. I would like @brian-brazil view on this one. I think that that error should not happen in real situation. If that can happen in real life, we might trigger a 1.15.3 release.

From my understanding however, this is more about crafting http requests that don't go over http. In which case at the moment you craft the request, you could manually set a RemoteAddr, e.g. 127.0.0.2:0 or something like that.

**Comment by pstibrany** [comment]
> Is there a real-life situation where that can error?

This field is documented to have undefined format, and only uses “IP:port” when request object is created by HTTP server in net/http package.

In our case, we construct request objects manually (based on requests in the frontend queue), and pass them to prometheus v1.API instance. As we don’t keep RemoteAddr, we need to fake this value now, otherwise API instance complains.

**Comment by brian-brazil** [comment]
> This field is documented to have undefined format, and only uses “IP:port” when request object is created by HTTP server in net/http package.

So that's a no then in Prometheus terms. Having the code simpler and more resilient doesn't hurt though.

**Comment by pstibrany** [comment]
> So that's a no then in Prometheus terms. Having the code simpler and more resilient doesn't hurt though.

I understand that Prometheus doesn't need this change, but given the size, code simplification (why should context creation fail?) and the fact that it helps your Cortex friends, perhaps we can still get it in?

**Comment by roidelapluie** [comment]
> > So that's a no then in Prometheus terms. Having the code simpler and more resilient doesn't hurt though.
> 
> I understand that Prometheus doesn't need this change, but given the size, code simplification (why should context creation fail?) and the fact that it helps your Cortex friends, perhaps we can still get it in?

Do we want to run a query if we are not able to log the client IP? Is there a security implication?

**Comment by pstibrany** [comment]
> Do we want to run a query if we are not able to log the client IP? Is there a security implication?

I assume this question was not for me. Just a note about client IPs -- what you really get here is possibly the address of last intermediate proxy, and not real client IP, so it's already best-effort only.

**Comment by pstibrany** [comment]
> But indeed we are changing logic bit here: Maybe someone relies on clientIP?

In Prometheus, `RemoteAddr` will always be in IP:port form, so there should be no change here.

**Comment by brian-brazil** [comment]
> I think @brian-brazil meant that we are happy to have this in, even though it does not help Prometheus in current moment (:

Yes, that's my meaning. This was actually a niggle when I was reviewing it, but I didn't bring it up.

**Pull request merged**